api/src/main/java/com/community/api/endpoint/avisoft/controller/Qualification/QualificationDetailsController.java

```diff
@@ -24,21 +27,33 @@ public class QualificationDetailsController
     protected QualificationDetailsService qualificationDetailsService;
     protected ExceptionHandlingImplement exceptionHandling;
     private ResponseService responseService;
-    public QualificationDetailsController(QualificationDetailsService qualificationDetailsService, ExceptionHandlingImplement exceptionHandling, ResponseService responseService)
+    private JwtUtil jwtTokenUtil;
+    private RoleService roleService;
+    public QualificationDetailsController(QualificationDetailsService qualificationDetailsService, ExceptionHandlingImplement exceptionHandling, ResponseService responseService,JwtUtil jwtTokenUtil, RoleService roleService)
     {
         this.qualificationDetailsService = qualificationDetailsService;
         this.exceptionHandling=exceptionHandling;
         this.responseService = responseService;
+        this.jwtTokenUtil = jwtTokenUtil;
+        this.roleService = roleService;
     }
 
     @PostMapping("/add/{customCustomerId}")
-    public ResponseEntity<?> addQualificationDetail(@PathVariable Long customCustomerId , @Valid @RequestBody QualificationDetails qualificationDetails) throws EntityAlreadyExistsException, ExaminationDoesNotExistsException, CustomerDoesNotExistsException {
+    public ResponseEntity<?> addQualificationDetail(@PathVariable Long customCustomerId , @Valid @RequestBody QualificationDetails qualificationDetails, @RequestHeader(value = "Authorization") String authHeader) throws EntityAlreadyExistsException, ExaminationDoesNotExistsException, CustomerDoesNotExistsException {
```

Comment:
here if someone enters string
give NumberFormatException exception

use this
Long id = Long.valueOf(customerId);

catch (NumberFormatException e) {
    return ResponseService.generateErrorResponse("Invalid customerId: expected a Long", HttpStatus.BAD_REQUEST);
}